### PR TITLE
Mode issues

### PR DESCRIPTION
**Pull request by nightwing**
@lennartcl

### CODE REVIEW

File: lib/ace/mode/batchfile_highlight_rules.js [hunk 1]
```diff
@@ -46,7 +46,7 @@ var BatchFileHighlightRules = function() {
...
...|eventcreate|eventquery.vbs|eventtriggers|expand|explorer|extract|exe2bin|fastopen|fc|fdisk|filever|find|findramd|findstr|for|forcedos|forfiles|format|fsutil|ftp|ftype|getmac|gettype|gpresult|graftabl|gwbasic|help|icacls|iexpress|if|ifconfig|inuse|ipconfig|jt|keyb|label|lfnfor|lh|loadhigh|lock|logman|logoff|makecab|md|mem|mkdir|mklink|mode|more|mountvol|move|msd|msg|msiexec|msinfo32|nbtstat|net|netdom|netsh|netstat|nltest|notepad|nslookup|ntbackup|ocsetup|odbcconf|openfiles|path|pathping|pause|pentnt|ping|ping6|pkgmgr|popd|powercfg|print|printui.exe|prncfg.vbs|prndrvr.vbs|prnjobs.vbs|prnmngr.vbs|prnport.vbs|prnqctl.vbs|prompt|pubprn.vbs|pushd|qappsrv|qbasic|qchain|qfarm|qprocess|qserver|query|quser|qwinsta|rasdial|rasphone|rd|readline|recimg|recover|reg|regedit|regedt32|regini|regsvr32|rem|ren|rename|replace|reset session|restore|rmdir|robocopy|route|runas|rundll|rundll32|sc|schtasks|secedit|set|setlocal|sfc|shadow|shell|shift|shutdown|sort|start|subinacl|subst|systeminfo|taskkill|tasklist|time|title|tracert|tracert6|tracerte|tree|truename|tscon|tsdiscon|tskill|tsprof|tsshutdn|type|tzchange|unlock|ver|verify|vhdmount|vol|winmsd|winmsdp|winsat|wmic|wscript|wupdmgr|xcacls|xcopy)\\b',
```
Comment (lennartcl) [inline lib/ace/mode/batchfile_highlight_rules.js]: `regedt32`? This seems like a fairly open-ended list? Should we have all of these?

File: lib/ace/mode/behaviour/cstyle.js [hunk 2]
```diff
@@ -250,14 +252,15 @@ var CstyleBehaviour = function(options) {
     });
 
     this.add("string_dquotes", "insertion", function(state, action, editor, session, text) {
-        if (text == '"' || text == "'") {
+        var quotes = session.$mode.$quotes || defaultQuotes
```
Comment (lennartcl) [inline lib/ace/mode/behaviour/cstyle.js]: semmy

File: lib/ace/mode/batchfile_highlight_rules.js (hunk 1 above)
Comment (nightwing) [inline lib/ace/mode/batchfile_highlight_rules.js]: @lennartcl i am not sure, was trying to add keywords from https://github.com/ajaxorg/ace/pull/3238, will revert it

**Comment by w3suli** [comment]
Almost good, but the percentage sign is still wrong.
eae38dc6-2876-11e7-9ad1-15d98e1822e0

([https://github.com/ajaxorg/ace/issues/3270](url))

**Comment by w3suli** [comment]
I found the cause of the mistake! This version and original version is bad.
bug
Recognition of blocks bounded by brackets is bad!

**Comment by nightwing** [comment]
@w3suli thanks for checking this, the new version should fix the issue above and https://github.com/ajaxorg/ace-builds/issues/106

**Comment by w3suli** [comment]
[https://rawgit.com/ajaxorg/ace/mode-issues/kitchen-sink.html](url)

Block management is OK :).
Only comment highlighting is incorrect, but there is another error generated by comment.

new bug

Important comment bug:
new comment bug

**Comment by nightwing** [comment]
@w3suli thanks, this should be fixed now

**Comment by w3suli** [comment]
Yes and implement it in the html module.
I'm glad I could help.

HTML:
html

**Review by macotouenaca — COMMENTED** [review]
ありがとうございます

**Pull request merged**